gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineClassUniquenessPluginIntegrationTest.groovy

```diff
@@ -0,0 +1,154 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline
+
+import java.nio.file.Files
+import java.util.stream.Stream
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.TaskOutcome
+
+class BaselineClassUniquenessPluginIntegrationTest extends AbstractPluginTest {
+
+    def standardBuildFile = """
+        plugins {
+            id 'java'
+            id 'com.palantir.baseline-class-uniqueness'
+        }
+        subprojects {
+            apply plugin: 'java'
+        }
+        repositories {
+            mavenCentral()
+            maven { url 'https://dl.bintray.com/palantir/releases' }
+        }
+    """.stripIndent()
+
+    def 'Task should run as part of :check'() {
+        when:
+        buildFile << standardBuildFile
+
+        then:
+        def result = with('check', '--stacktrace').build()
+        result.task(':checkClassUniqueness').outcome == TaskOutcome.SUCCESS
+    }
+
+    def 'detect duplicates in two external jars'() {
+        when:
+        buildFile << standardBuildFile
+        buildFile << """
+        dependencies {
+            compile group: 'javax.el', name: 'javax.el-api', version: '3.0.0'
+            compile group: 'javax.servlet.jsp', name: 'jsp-api', version: '2.1'
+            compile 'com.google.code.findbugs:annotations:3.0.0'
+            compile 'com.github.stephenc.jcip:jcip-annotations:1.0-1'
+            compile 'commons-logging:commons-logging:1.2'
+            compile 'org.slf4j:jcl-over-slf4j:1.7.25'
+            compile 'com.palantir.tritium:tritium-api:0.9.0'
+            compile 'com.palantir.tritium:tritium-core:0.9.0'
+        }   
+        """.stripIndent()
+        BuildResult result = with('checkClassUniqueness').buildAndFail()
+
+        then:
+        result.getOutput().contains("Identically named classes found in 2 jars ([javax.servlet.jsp:jsp-api:2.1, javax.el:javax.el-api:3.0.0]): [javax.")
+        result.getOutput().contains("'runtime' contains multiple copies of identically named classes")
+        result.getOutput().contains("(4 classes)   com.google.code.findbugs:annotations:3.0.1      net.jcip:jcip-annotations:1.0                   com.github.stephenc.jcip:jcip-annotations:1.0-1");
```

Comment:
I feel like people would want to know the class names that are duplicated -- thoughts on listing that out? (it would be more verbose =/)

Reply:
These appear in log lines (at ERROR) level, so they'll appear in the console